projects/ntlm2/fuzz_auth.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/python3
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import atheris
+import sys
+with atheris.instrument_imports():
+    import requests
+    from requests.exceptions import InvalidURL,ConnectionError
+    from requests_ntlm2 import HttpNtlmAuth
+
+def TestInput(data):
+    fdp = atheris.FuzzedDataProvider(data)
+
+    session = requests.Session()
+
+    auth = HttpNtlmAuth(fdp.ConsumeString(50),fdp.ConsumeString(10))
+    session.auth = HttpNtlmAuth(fdp.ConsumeString(50),fdp.ConsumeString(10))
+
+    try:
+       requests.get('http://localhost:%d/%s'%(fdp.ConsumeIntInRange(10000,65535),fdp.ConsumeString(20)),auth=auth)
```

Comment:
Does it make sense for this to hit a real server (on localhost) somehow?

Reply:
@arthurscchan I think we discussed this internally? Can we set up something similar to https://github.com/google/oss-fuzz/blob/master/projects/requests/fuzz_server.py

Reply:
Fuzzer changed with managed server